assets/js/utils/get-query.js

```diff
@@ -1,29 +1,38 @@
-export default function getQuery( attributes ) {
+export default function getQuery( attributes, name ) {
 	const { categories, columns, orderby, rows } = attributes;
 
 	const query = {
 		status: 'publish',
 		per_page: rows * columns,
 	};
 
-	if ( categories ) {
+	if ( categories && categories.length ) {
 		query.category = categories.join( ',' );
 	}
 
-	if ( 'price_desc' === orderby ) {
-		query.orderby = 'price';
-		query.order = 'desc';
-	} else if ( 'price_asc' === orderby ) {
-		query.orderby = 'price';
-		query.order = 'asc';
-	} else if ( 'title' === orderby ) {
-		query.orderby = 'title';
-		query.order = 'asc';
-	} else if ( 'menu_order' === orderby ) {
-		query.orderby = 'menu_order';
-		query.order = 'asc';
-	} else {
-		query.orderby = orderby;
+	if ( orderby ) {
+		if ( 'price_desc' === orderby ) {
+			query.orderby = 'price';
+			query.order = 'desc';
+		} else if ( 'price_asc' === orderby ) {
+			query.orderby = 'price';
+			query.order = 'asc';
+		} else if ( 'title' === orderby ) {
+			query.orderby = 'title';
+			query.order = 'asc';
+		} else if ( 'menu_order' === orderby ) {
+			query.orderby = 'menu_order';
+			query.order = 'asc';
+		} else {
+			query.orderby = orderby;
+		}
+	}
+
+	// Toggle shortcode atts depending on block type.
+	switch ( name ) {
+		case 'woocommerce/product-best-sellers':
+			query.orderby = 'popularity';
```

Comment:
Nitpicking a bit here, but could we remove this logic and pass the query param from the `ProductBestSellersBlock` component?

Reply:
No, I was suggesting that instead of passing the name and using a switch statement to modify the query, that we keep the `getQuery` util clean and instead pass the `query.orderby` to `getQuery()` from the `ProductBestSellersBlock` [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/221/files#diff-91be8d8a2c0062f08023c4b23cd0ef4eR58).

I think keeping the component logic out of the utility might make maintenance a bit easier long-term unless you see an advantage to modifying `orderby` inside of `getQuery`.

Reply:
We don't pass a query into `getQuery` though– we pass the block attributes directly (and now the block name). We'd need to introduce another param for the query, which IMO adds mental overhead to a function that is "get the query from the block attributes".